execute/aggregate.go

```diff
@@ -46,13 +47,33 @@ type AggregateTransformation interface {
 	Closer
 }
 
+// AggregateParallelTransformation is an AggregateTransformation that is capable of
+// processing chunks from within the same group key in parallel.
+//
+// The thing that differentiates this from a normal AggregateTransformation is having
+// multiple parents and the capability to merge two existing states into a single one
+// that will be passed to Compute.
+type AggregateParallelTransformation interface {
+	AggregateTransformation
+
+	// Merge will take two existing states produced by the Aggregate method and merge them
+	// into a single state.
+	Merge(into, from interface{}, mem memory.Allocator) (interface{}, error)
+}
+
 type aggregateTransformation struct {
 	t AggregateTransformation
 	d *TransportDataset
+
+	merge    func(into, from interface{}, mem memory.Allocator) (interface{}, error)
+	parents  map[DatasetID]*RandomAccessGroupLookup
+	finished int
+	err      error
+	mu       sync.RWMutex
```

Comment:
Would it be possible to make a separate `parallelAggregateTransformation`? The mixing of parallel and sequential concerns in the methods here makes it more difficult to read the code than necessary (IMO)

Reply:
Done. I think it's a bit cleaner now.